File: docs/snippets/quickstart.ts
// #region create
import { createZooworkClient } from '@zoowork-ai/sdk'

const client = createZooworkClient()
const agent = await client.createAgent({
  resource: { name: 'quickstart-agent' },
})
console.log(`Agent ID: ${agent.agent_id}`)
// #endregion create

// #region start
await client.startAgent(agent.agent_id)
// #endregion start

// #region session
const session = await client.createSession(agent.agent_id, {})
console.log(`Session ID: ${session.session_id}`)
// #endregion session

// #region send
const receipt = await client.postEvents(agent.agent_id, session.session_id, [{
  type: 'user.message',
  content: 'Create report.md with a sales table: January $100, February $120, March $80. '
    + 'Include the total. Read the file to verify it, then report the total.',
}])
if (receipt.events[0]?.accepted !== true) throw new Error('Message was not accepted')
// #endregion send

// #region stream
import { assistantText, toolCall, isRunFinished, runOutcome } from '@zoowork-ai/sdk'

const controller = new AbortController()
let outcome: ReturnType<typeof runOutcome>

try {
  for await (const event of client.streamEvents(agent.agent_id, session.session_id, {
    signal: controller.signal,
  })) {
    process.stdout.write(assistantText(event))
    const call = toolCall(event)
    if (call?.phase === 'start') console.log(`\n[tool] ${call.toolName}`)

    if (isRunFinished(event)) {
      outcome = runOutcome(event)
      console.log(`\nTurn: ${outcome}`)
      break
    }
  }
} finally {
  controller.abort()
}
if (outcome !== 'succeeded') throw new Error(`Turn did not succeed: ${outcome ?? 'stream closed'}`)
// #endregion stream

// #region cleanup
await client.stopAgent(agent.agent_id)
await client.deleteAgent(agent.agent_id)
// #endregion cleanup

File: docs/snippets/quickstart.py
# region create
import asyncio

from zoowork import (
    assistant_text,
    create_zoowork_client,
    is_run_finished,
    run_outcome,
    tool_call,
)


async def main() -> None:
    async with create_zoowork_client() as client:
        agent = await client.create_agent({"name": "quickstart-agent"})
        agent_id = str(agent["agent_id"])
        print(f"Agent ID: {agent_id}")
# endregion create

# region start
# Continue inside main().
        await client.start_agent(agent_id)
# endregion start

# region session
# Continue inside main().
        session = await client.create_session(agent_id)
        session_id = str(session["session_id"])
        print(f"Session ID: {session_id}")
# endregion session

# region send
# Continue inside main().
        receipt = await client.post_events(
            agent_id,
            session_id,
            [{
                "type": "user.message",
                "content": (
                    "Create report.md with a sales table: January $100, February $120, "
                    "March $80. Include the total. Read the file to verify it, then report "
                    "the total."
                ),
            }],
        )
        if not receipt or receipt[0].get("accepted") is not True:
            raise RuntimeError("Message was not accepted")
# endregion send

# region stream
# Continue inside main().
        outcome = None
        async for event in client.stream_events(agent_id, session_id):
            print(assistant_text(event), end="", flush=True)
            call = tool_call(event)
            if call is not None and call.phase == "start":
                print(f"\n[tool] {call.tool_name}")

            if is_run_finished(event):
                outcome = run_outcome(event)
                print(f"\nTurn: {outcome}")
                break

        if outcome != "succeeded":
            raise RuntimeError(f"Turn did not succeed: {outcome or 'stream closed'}")
# endregion stream

# region cleanup
# Finish inside main(), then run it.
        await client.stop_agent(agent_id)
        await client.delete_agent(agent_id)


if __name__ == "__main__":
    asyncio.run(main())
# endregion cleanup
